pkg/kube/inject/inject.go

```diff
@@ -572,6 +572,17 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, typeMetadata *
 		"toLower":             strings.ToLower,
 	}
 
+	// Allows the template to use env variables from istiod.
+	// Istiod will use a custom template, without 'values.yaml', and the pod will have
+	// an optional 'vendor' configmap where additional settings can be defined.
+	funcMap["env"] = func(key string, def string) string {
+		val := os.Getenv(key)
```

Comment:
Do we ever want to set an env var to `""` explicitly? if so, we should use lookup env which can distinguish unset and `""`

Reply:
When we find a use case - we could. But by that time - the mesh API will be in place, and we can replace this code with the equivalent ( we'll still need injector to process the 'vendor' envs, but 
will be placed on overrides of mesh.proxyConfig.env )